.github/actions/publish/action.yml

```diff
@@ -0,0 +1,87 @@
+---
+name: publish binaries
+description: package GitHub release for node-pre-gyp
+# https://github.com/product-os/flowzone/tree/master/.github/actions
+inputs:
+  json:
+    description: 'JSON stringified object containing all the inputs from the calling workflow'
+    required: true
+  secrets:
+    description: 'JSON stringified object containing all the secrets from the calling workflow'
+    required: true
+  variables:
+    description: 'JSON stringified object containing all the variables from the calling workflow'
+    required: true
+  # --- custom environment
+  NODE_VERSION:
+    description: 'Node.js version'
+    default: '18.x'
+
+runs:
+  # https://docs.github.com/en/actions/creating-actions/creating-a-composite-action
+  using: 'composite'
+  steps:
+    - name: Setup Node.js
+      uses: actions/setup-node@v3
+      with:
+        node-version: ${{ inputs.NODE_VERSION }}
+        registry-url: 'https://registry.npmjs.org'
+
+    - name: Build JSON array from pipe separated list
+      id: config
+      uses: kanga333/json-array-builder@c7cd9d3a8b17cd368e9c2210bc3c16b0e2714ce5 # v0.2.1
+      with:
+        cmd: bash -c "echo $INPUT | tr -d '[:space:]'"
+        separator: "|"
+      env:
+        INPUT: ${{ env.matrix_value }}
+
+    - name: Login to GitHub Container Registry
+      continue-on-error: true
+      uses: docker/login-action@343f7c4344506bcbf9b4de18042ae17996df046d # v3.0.0
+      with:
+        registry: ghcr.io
+        username: ${{ github.actor }}
+        password: ${{ fromJSON(inputs.secrets).GITHUB_TOKEN }}
+
+    - name: Copy the binary from the built docker image
+      shell: bash
+      run: |
+        set -ex
+
+        image="ghcr.io/balena-io-modules/node-systemd:${{ github.event.pull_request.head.sha }}"
+        while ! docker buildx imagetools inspect "${image}" >/dev/null 2>&1
+        do
+          echo "waiting for ${image}..."
+          sleep $(((RANDOM % 10) + 10))
+        done
+
+        # Copy the binary from the image to the local folder for packaging
+        container="node-systemd-$ARCH"
+        docker pull --platform "$PLATFORM" "${image}"
+        docker create --platform "$PLATFORM" --name "${container}" "${image}"
+        docker cp "${container}:/usr/src/app/native" ./native
+        docker rm "${container}"
+        docker rmi -f "${image}"
+      env:
+        PLATFORM: ${{ fromJSON(steps.config.outputs.build)[0] }}
+        ARCH: ${{ fromJSON(steps.config.outputs.build)[1] }}
+
+    - name: Package the target
+      shell: bash
+      # This will make a node-pre-gyp package.
+      run: npx node-pre-gyp package --target_libc=$LIBC --target_platform=linux --target_arch=$ARCH
```

Comment:
Why isn't this done as part of the Docker build?

Reply:
Hmm, good point, no reason other than I didn't think of it. If node-pre-gyp can detect the architecture and platform from the container that might be better, but if I have to parse the `arch` I'm not sure that will be an improvement. I'll test it